### ISSUE TITLE
Compile errors on linux(debian)  ARM environment

### ISSUE THREAD

**Issue by elmerubalde** [Azure/azure-iot-sdk-c#316]
- **OS and version used:** Debian GNU/Linux 8 

- **SDK version used:** 2017-11-17 and up 

# Description of the issue:

Tried on 2017-11-17 and the latest Git repo.
Unable to build. Fails because of missing  parson and socket_async_os.
# Code sample exhibiting the issue:

# Console log of the issue:
Scanning dependencies of target umqtt
[ 51%] Building C object umqtt/CMakeFiles/umqtt.dir/src/mqtt_client.c.o
[ 52%] Building C object umqtt/CMakeFiles/umqtt.dir/src/mqtt_codec.c.o
[ 53%] Building C object umqtt/CMakeFiles/umqtt.dir/src/mqtt_message.c.o
Linking C static library libumqtt.a
[ 53%] Built target umqtt
Scanning dependencies of target iothub_client
[ 54%] Building C object iothub_client/CMakeFiles/iothub_client.dir/src/iothub_client.c.o
[ 55%] Building C object iothub_client/CMakeFiles/iothub_client.dir/src/version.c.o
[ 56%] Building C object iothub_client/CMakeFiles/iothub_client.dir/src/iothubtransport.c.o
Linking C static library libiothub_client.a
[ 56%] Built target iothub_client
Scanning dependencies of target iothub_client_mqtt_transport
[ 57%] Building C object iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/version.c.o
[ 58%] Building C object iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/iothub_client_authorization.c.o
[ 59%] Building C object iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/iothub_message.c.o
[ 60%] Building C object iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/iothub_client_ll.c.o
[ 61%] Building C object iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/blob.c.o
[ 62%] Building C object iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/iothubtransport_mqtt_common.c.o
/tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c: In function 'mqtt_operation_complete_callback':
/tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c:1431:31: error: too few arguments to function 'mqtt_client_disconnect'
 (void)mqtt_client_disconnect(transport_data->mqttClient);
       ^
In file included from /tmp/azure-iot-sdk-c/umqtt/inc/azure_umqtt_c/mqtt_client.h:18:0,
     from /tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c:17:
/tmp/azure-iot-sdk-c/umqtt/inc/azure_umqtt_c/mqtt_client.h:54:26: note: declared here
 MOCKABLE_FUNCTION(, int, mqtt_client_disconnect, MQTT_CLIENT_HANDLE, handle, ON_MQTT_DISCONNECTED_CALLBACK, callback, void*, ctx);
  ^
/tmp/azure-iot-sdk-c/umqtt/deps/c-utility/inc/azure_c_shared_utility/umock_c_prod.h:31:22: note: in definition of macro 'MOCKABLE_FUNCTION'
     result modifiers function(IF(COUNT_ARG(__VA_ARGS__),,void) FOR_EACH_2_COUNTED(UMOCK_C_PROD_ARG_IN_SIGNATURE, __VA_ARGS__));
          ^
/tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c: In function 'DisconnectFromClient':
/tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c:1778:11: error: too few arguments to function 'mqtt_client_disconnect'
     (void)mqtt_client_disconnect(transport_data->mqttClient);
           ^
In file included from /tmp/azure-iot-sdk-c/umqtt/inc/azure_umqtt_c/mqtt_client.h:18:0,
     from /tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c:17:
/tmp/azure-iot-sdk-c/umqtt/inc/azure_umqtt_c/mqtt_client.h:54:26: note: declared here
 MOCKABLE_FUNCTION(, int, mqtt_client_disconnect, MQTT_CLIENT_HANDLE, handle, ON_MQTT_DISCONNECTED_CALLBACK, callback, void*, ctx);
  ^
/tmp/azure-iot-sdk-c/umqtt/deps/c-utility/inc/azure_c_shared_utility/umock_c_prod.h:31:22: note: in definition of macro 'MOCKABLE_FUNCTION'
     result modifiers function(IF(COUNT_ARG(__VA_ARGS__),,void) FOR_EACH_2_COUNTED(UMOCK_C_PROD_ARG_IN_SIGNATURE, __VA_ARGS__));
          ^
/tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c: In function 'InitializeConnection':
/tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c:1846:27: error: too few arguments to function 'mqtt_client_disconnect'
         (void)mqtt_client_disconnect(transport_data->mqttClient);
   ^
In file included from /tmp/azure-iot-sdk-c/umqtt/inc/azure_umqtt_c/mqtt_client.h:18:0,
     from /tmp/azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c:17:
/tmp/azure-iot-sdk-c/umqtt/inc/azure_umqtt_c/mqtt_client.h:54:26: note: declared here
 MOCKABLE_FUNCTION(, int, mqtt_client_disconnect, MQTT_CLIENT_HANDLE, handle, ON_MQTT_DISCONNECTED_CALLBACK, callback, void*, ctx);
  ^
/tmp/azure-iot-sdk-c/umqtt/deps/c-utility/inc/azure_c_shared_utility/umock_c_prod.h:31:22: note: in definition of macro 'MOCKABLE_FUNCTION'
     result modifiers function(IF(COUNT_ARG(__VA_ARGS__),,void) FOR_EACH_2_COUNTED(UMOCK_C_PROD_ARG_IN_SIGNATURE, __VA_ARGS__));
          ^
iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/build.make:169: recipe for target 'iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/iothubtransport_mqtt_common.c.o' failed
make[2]: *** [iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/src/iothubtransport_mqtt_common.c.o] Error 1
CMakeFiles/Makefile2:1169: recipe for target 'iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/all' failed
make[1]: *** [iothub_client/CMakeFiles/iothub_client_mqtt_transport.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by ewertons** [comment]
Hi @elmerubalde ,
this environment setup is covered in our gated builds, and we did not have any failures there recently.

you mentioned the failures are due to parson and "socket_async_os" not being installed? that is not clear on the traces you shared. Could you attach the full compilation log in this issue please? Including the command line run to compile the solution.

**Comment by elmerubalde** [comment]
Hi ewertons, 

Thanks for checking.

Good news is I wasn't able to reproduce the same error today.
I think some dependencies failed to download during my build process. 

The commands I'm using were:

sudo apt-get update
sudo apt-get install -y --no-install-recommends build-essential
sudo apt-get install -y libcurl4-openssl-dev libssl-dev uuid-dev git cmake curl
git clone --recursive https://github.com/Azure/azure-iot-sdk-c.git  

cd azure-iot-sdk-c
./build_all/linux/build.sh --no-http

I tried the process a few times today and looks solid.
So, looks like error was caused by my environment .

Thanks again.
